### PR TITLE
[orchagent]:add -t time option to set mac aging time when orchagent is initializating

### PR DESCRIPTION
**Pull request by gongjianLhr**
Signed-off-by: Haoran Liu gongjian.lhr@alibaba-inc.com

### CODE REVIEW

File: orchagent/main.cpp [hunk 1]
```diff
@@ -104,10 +104,10 @@ int main(int argc, char **argv)
 
     int opt;
     sai_status_t status;
-
+    bool gMacAgingTimeSet = false;
```
Comment (stcheng) [inline orchagent/main.cpp]: for this attribute, i think there's no need to make it global.

File: orchagent/main.cpp [hunk 2]
```diff
@@ -149,6 +149,20 @@ int main(int argc, char **argv)
...
+            int tmptime = stoi(optarg);
+            if (tmptime < 0)
+            {   
+                macAgingTime = 0;
+                SWSS_LOG_DEBUG("Input macAgingTime < 0, set macAgingTime to 0");
+            }
+            else
+                macAgingTime = tmptime;
+        }
+            break;
```
Comment (stcheng) [inline orchagent/main.cpp]: why this break is out of the case's scope?

File: orchagent/main.cpp [hunk 3]
```diff
@@ -149,6 +149,20 @@ int main(int argc, char **argv)
...
             }
             break;
+        case 't':
+        {
+            gMacAgingTimeSet = true;
+            int tmptime = stoi(optarg);
+            if (tmptime < 0)
+            {   
+                macAgingTime = 0;
+                SWSS_LOG_DEBUG("Input macAgingTime < 0, set macAgingTime to 0");
```
Comment (stcheng) [inline orchagent/main.cpp]: use SWSS_LOG_WARN instead

Comment (stcheng) [inline orchagent/main.cpp]: if we don't provide the parameter, what's the default value. do we need to explicitly set the value?

**Comment by lguohan** [comment]
is there a requirement to change the mac aging timer without restart orchagent? if yes, then need to go with configdb.

File: orchagent/main.cpp (hunk 1 above)
Comment (gongjianLhr) [reply orchagent/main.cpp]: Hi Stcheng, thanks for review, yes there is no need to make it global. I just named it wrongly.

File: orchagent/main.cpp (hunk 3 above)
Comment (gongjianLhr) [reply orchagent/main.cpp]: We don't need to set the default value explicitly. SAI has set the default value in ocp_sai/include/sai/saiswitch.h :
/*
    * Dynamic FDB entry aging time in seconds [sai_uint32_t]
    *   Zero means aging is disabled.
    *  (default to zero)
*/

File: orchagent/main.cpp (hunk 2 above)
Comment (gongjianLhr) [reply orchagent/main.cpp]: There is no difference for the break to keep inside or outside the brace .
I added the brace only to handle the situation of local variable tmptime.

**Comment by gongjianLhr** [comment]
Hi lguohan, thanks for review. For now, we don't have the scenario that needs to change the mac aging time dynamically.

**Comment by stcheng** [comment]
Hello @gongjianLhr , i'll add a patch to dynamically change the switch attribute. Once my fix is done, you could add this patch upon that, which will allow changing mac aging time dynamically.

**Comment by gongjianLhr** [comment]
@stcheng that's great. when will this patch be available?

**Comment by stcheng** [comment]
@gongjianLhr i'll working on that. it will be available soon after we merge the 1.0.3 branch into the master - that is to say, within 1-2 weeks.

**Comment by stcheng** [comment]
@gongjianLhr could you take a look at this pull request?
https://github.com/Azure/sonic-swss/pull/307
Later we could add the MAC aging time attribute.

**Comment by stcheng** [comment]
Hi @gongjianLhr  could you close this pull request and open a new one using the new switch orch architecture.

**Comment by gongjianLhr** [comment]
OK

**Pull request closed without merging**